src/test/ui/consts/const-mut-refs/mut_ref_in_final.rs

```diff
@@ -0,0 +1,24 @@
+#![feature(const_mut_refs)]
+#![feature(const_fn)]
+#![feature(raw_ref_op)]
+const NULL: *mut i32 = std::ptr::null_mut();
+const A: *const i32 = &4;
+
+// It could be made sound to allow it to compile,
+// but we do not want to allow this to compile,
+// as that would be an enormous footgun in oli-obk's opinion.
+const B: *mut i32 = &mut 4; //~ ERROR untyped pointers are not allowed in constant
```

Comment:
I think this is exactly my main concern here... there are legitimate contsants that are disallowed by the same check, such as
```rust
const CONST_RAW: *const Vec<i32> = &Vec::new() as *const _;
```
So your proposed PR here makes it very very hard to ever allow `CONST_RAW`. It is basically closing a door.

Reply:
I'm currently looking into creating static checks similar to the interior mutability checks. This spawned off https://github.com/rust-lang/rust/pull/78679.